### PR TITLE
Revert "remove dnsprovider from supported extension kinds" temporarily

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area control-plane
/kind cleanup

**What this PR does / why we need it**:
This PR reverts the PR #6840 "Remove DNSProvider from supported extension kinds" temporarily, as it prevents removing the `dns-external` extension if it was not removed before updating to Gardener `v1.58.0`.
It should be reapplied for `v1.60.0` or later.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
This reverts commit c18cd01b5bf17f4d8fe1abf622b25b8064b5df95.

**Release note**:

```other operator
Revert removal `DNSProvider` from supported extension kinds until `v1.60.0` or later.
```

### CODE REVIEW

File: pkg/apis/core/validation/controllerregistration.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ var SupportedExtensionKinds = sets.NewString(
 	extensionsv1alpha1.NetworkResource,
 	extensionsv1alpha1.OperatingSystemConfigResource,
 	extensionsv1alpha1.WorkerResource,
+	// TODO: drop this with v1.55 or later to still support removal of any controller registrations specifying resources of kind DNSProvider, see https://github.com/gardener/gardener/issues/5270
```
Comment (dguendisch) [inline pkg/apis/core/validation/controllerregistration.go]: maybe adapt the comment?

**Comment by rfranzke** [comment]
We need this PR because we were not following our own release notes? 🤣

**Comment by shafeeqes** [comment]
>  @MartinWeindel: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-gardener-integration	[7f54006](https://github.com/gardener/gardener/commit/7f540065d5a8bd25eb019a21892eae05ab05de1d)	[link](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/6951/pull-gardener-integration/1588145125026959360)	true	`/test pull-gardener-integration`
> [Full PR test history](https://prow.gardener.cloud/pr-history?org=gardener&repo=gardener&pr=6951). [Your PR dashboard](https://prow.gardener.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3AMartinWeindel). [Command help for this repository](https://prow.gardener.cloud/command-help?repo=gardener%2Fgardener). Please help us cut down on flakes by linking this test failure to an [open flake report](https://github.com/gardener/gardener/issues?q=is:issue+is:open+label:kind/flake) or filing a [new flake report](https://github.com/gardener/gardener/issues/new?template=flaking-test.md&title=%5BFlaky+Test%5D+FLAKING+TEST%2FSUITE) if you can't find an existing one. Also see our [testing guideline](https://github.com/gardener/gardener/blob/master/docs/development/testing.md) for how to avoid and hunt flakes.

Due to #6928 
/test pull-gardener-integration

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**